File: 03_TestConvertersApp/Backend_dotnet/Pipeline.Storage/Utils/StorageUtils.cs
using System.IO;

namespace Pipeline.Storage.Utils
{
    public class StorageUtils
    {
        public static string MakeOriginalFilePath(string jobId, string key)
        {
            return $"{jobId}/{key}/Original{Path.GetExtension(key)}";
        }
        
        public static string MakeConvertedFilePath(string jobId, string key, string converter, string convertedExtension)
        {
            return $"{jobId}/{key}/{converter}{convertedExtension}";
        }
    }
}

File: 03_TestConvertersApp/Backend_node/lib/storageUtils.js
function makeOriginalFilePath(jobId, key) {
    const extension = key.split('.').pop(); 
    return `${jobId}/${key}/Original.${extension}`
}

function makeConvertedFilePath(jobId, key, converter, convertedExtension) {
    return `${jobId}/${key}/${converter}${convertedExtension}`
}

function makeConvertedAdditionalFilePath(jobId, key, converter, fileName) {
    return `${jobId}/${key}/${converter}_Additional/${fileName}`
}

module.exports = {
    makeOriginalFilePath: makeOriginalFilePath,
    makeConvertedFilePath: makeConvertedFilePath,
    makeConvertedAdditionalFilePath: makeConvertedAdditionalFilePath
  }